MobiusTest/Source/NextMatchers.swift

```diff
@@ -132,9 +132,48 @@ public func hasEffects<Model, Effect: Equatable>(
     line: UInt = #line
 ) -> NextPredicate<Model, Effect> {
     return { (next: Next<Model, Effect>) in
-        let actual = next.effects
-        if !expected.allSatisfy(actual.contains) {
-            return .failure(message: "Expected <\(actual)> to contain <\(expected)>", file: file, line: line)
+        if !expected.allSatisfy(next.effects.contains) {
+            return .failure(message: "Expected <\(next.effects)> to contain <\(expected)>", file: file, line: line)
+        }
+        return .success
+    }
+}
+
+/// Constructs a matcher that matches if only the supplied effects are present in the supplied `Next`, in any order.
+///
+/// - Parameter expected: the effects to match (possibly empty)
+/// - Returns: a `Predicate` that matches `Next` instances that include all the supplied effects
+public func hasOnlyEffects<Model, Effect: Equatable>(
+    _ expected: [Effect],
+    file: StaticString = #file,
+    line: UInt = #line
+) -> NextPredicate<Model, Effect> {
+    return { (next: Next<Model, Effect>) in
+        var unmatchedActual = next.effects
+        var unmatchedExpected = expected
+        zip(next.effects, expected).forEach {
+            _ = unmatchedActual.firstIndex(of: $1).map { unmatchedActual.remove(at: $0) }
+            _ = unmatchedExpected.firstIndex(of: $0).map { unmatchedExpected.remove(at: $0) }
+        }
```

Comment:
@kmcbride I'm trying to finalize #177 (finally!) and this code is causing me troubles. When for example `next.effects == [.a, .b]` and `expected == [.b]`, then after running this code `unmatchedExpected == [.b]`, which I think is incorrect, as `.b` could actually be matched.

Could this be implemented this way (which would in my opinion fix the problem):
```
let unmatchedActual = next.effects.filter { !expected.contains($0) }
let unmatchedExpected = expected.filter { !next.effects.contains($0) }
```

or is there any edge case I'm not seeing here?

Reply:
The intention here was to do the minimum work necessary to determine if the expectation could be satisfied, so the arrays don't always contain the "true" diff between the two inputs even though the returned value is correct. Feel free to adjust as needed for computing the actual diff 😬